server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataReaderByTimestamp.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.query.reader.seriesRelated;
+
+import java.io.IOException;
+import java.util.Objects;
+import org.apache.iotdb.db.engine.querycontext.QueryDataSource;
+import org.apache.iotdb.db.query.context.QueryContext;
+import org.apache.iotdb.db.query.reader.IReaderByTimestamp;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.file.metadata.statistics.Statistics;
+import org.apache.iotdb.tsfile.read.common.BatchData;
+import org.apache.iotdb.tsfile.read.common.Path;
+import org.apache.iotdb.tsfile.read.filter.TimeFilter;
+import org.apache.iotdb.tsfile.read.filter.basic.Filter;
+
+
+public class SeriesDataReaderByTimestamp extends AbstractDataReader implements
+    IReaderByTimestamp {
+
+  private BatchData batchData;
+  private Filter filter;
+
+  public SeriesDataReaderByTimestamp(Path seriesPath, TSDataType dataType, QueryContext context,
+      QueryDataSource dataSource) {
+    super(seriesPath, dataType, context, dataSource.getSeqResources(),
+        dataSource.getUnseqResources());
+  }
+
+  @Override
+  public Object getValueInTimestamp(long timestamp) throws IOException {
+    this.filter = TimeFilter.gtEq(timestamp);
+    if (batchData == null || batchData.getTimeByIndex(batchData.length() - 1) < timestamp) {
+      if (!hasNext(timestamp)) {
+        return null;
+      }
+    }
+
+    return batchData.getValueInTimestamp(timestamp);
+  }
+
+  @Override
+  public boolean hasNext() throws IOException {
```

Comment:
could this method be removed from the IReaderByTimestamp?

Reply:
I'm not going to fix it in this pr